src/VirtoCommerce.ProfileExperienceApiModule.Data/Commands/RegisterCompanyCommandHandler.cs

```diff
@@ -0,0 +1,194 @@
+using VirtoCommerce.CustomerModule.Core.Model;
+using VirtoCommerce.CustomerModule.Core.Services;
+using VirtoCommerce.ExperienceApiModule.Core.Models;
+using VirtoCommerce.ExperienceApiModule.Core.Services;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.Platform.Core.DynamicProperties;
+using VirtoCommerce.Platform.Core.GenericCrud;
+using VirtoCommerce.Platform.Core.Security;
+using VirtoCommerce.Platform.Core.Settings;
+using VirtoCommerce.ProfileExperienceApiModule.Data.Aggregates;
+using VirtoCommerce.StoreModule.Core.Model;
+using VirtoCommerce.StoreModule.Core.Services;
+using VirtoCommerce.NotificationsModule.Core.Extensions;
+using VirtoCommerce.NotificationsModule.Core.Services;
+using VirtoCommerce.CustomerModule.Core.Notifications;
+
+namespace VirtoCommerce.ProfileExperienceApiModule.Data.Commands
+{
+    public class RegisterCompanyCommandHandler : IRequestHandler<RegisterCompanyCommand, RegisterCompanyAggregate>
+    {
+        private readonly IMapper _mapper;
+        private readonly IDynamicPropertyUpdaterService _dynamicPropertyUpdater;
+        private readonly IMemberService _memberService;
+        private readonly Func<UserManager<ApplicationUser>> _userManagerFactory;
+        private readonly IStoreNotificationSender _storeNotificationSender;
+        private readonly RoleManager<Role> _roleManager;
+        private readonly ICrudService<Store> _storeService;
+        private readonly INotificationSearchService _notificationSearchService;
+        private readonly INotificationSender _notificationSender;
+
+        private const string Creator = "frontend";
+        private const string UserType = "Manager";
+        private const string MaintainerRoleId = "org-maintainer";
+
+        public RegisterCompanyCommandHandler(IMapper mapper,
+            IDynamicPropertyUpdaterService dynamicPropertyUpdater,
+            IMemberService memberService,
+            Func<UserManager<ApplicationUser>> userManagerFactory,
+            IStoreNotificationSender storeNotificationSender,
+            RoleManager<Role> roleManager,
+            ICrudService<Store> storeService,
+            INotificationSearchService notificationSearchService,
+            INotificationSender notificationSender)
+        {
+            _mapper = mapper;
+            _dynamicPropertyUpdater = dynamicPropertyUpdater;
+            _memberService = memberService;
+            _userManagerFactory = userManagerFactory;
+            _storeNotificationSender = storeNotificationSender;
+            _roleManager = roleManager;
+            _storeService = storeService;
+            _notificationSearchService = notificationSearchService;
+            _notificationSender = notificationSender;
+        }
+
+        public virtual async Task<RegisterCompanyAggregate> Handle(RegisterCompanyCommand request,
+            CancellationToken cancellationToken)
+        {
+            var aggregate = new RegisterCompanyAggregate();
+
+            try
```

Comment:
Use `CancellationToken` to implement rollback flow, remove try..catch and unnecessary `Exceptions`. 
Pseudocode:
```
await RegisterCompanyAsync(cancellationToken);
if (cancellationToken.IsCancelled)
    await RollBackMembersCreationAsync();
```